packages/taco/src/taco-config-validator.ts

```diff
@@ -0,0 +1,362 @@
+ */
+export function validateDomainChain(config: DomainChainConfig): boolean {
+  const domainInfo = Object.values(DOMAINS).find(
+    (info) => info.domain === config.domain,
+  );
+  return !!domainInfo && domainInfo.chainId === config.chainId;
+}
+
+/**
+ * Generic validation result interface
+ */
+export interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+/**
+ * TACo Configuration Validator
+ *
+ * Validates TACo client configurations, domains, and provider compatibility.
+ * Provides both fast and full validation methods for TACo operations.
+ */
+export class TacoConfigValidator {
+  /**
+   * Get all supported TACo domain names
+   * @returns {DomainName[]} Array of supported TACo domain names ('lynx', 'tapir', 'mainnet')
+   */
+  static getSupportedDomains(): DomainName[] {
+    return Object.values(DOMAINS).map((domain) => domain.domain);
+  }
+
+  /**
+   * Check if domain is valid
+   * @param {DomainName} domain - TACo domain name to check ('lynx', 'tapir', 'mainnet')
+   * @returns {boolean} True if domain exists
+   */
+  static isValidDomain(domain: DomainName): boolean {
+    return !!domain && this.getSupportedDomains().includes(domain);
+  }
+
+  /**
+   * Get expected chain ID for domain from DOMAINS configuration
+   * @param {DomainName} domain - Domain name to look up
+   * @returns {number | undefined} Chain ID for the domain, undefined if not found
+   * @private
+   */
+  private static getExpectedChainId(domain: DomainName): number | undefined {
+    const domainEntry = Object.values(DOMAINS).find(
+      (domainConfig) => domainConfig.domain === domain,
+    );
+    return domainEntry?.chainId;
+  }
+
+  /**
+   * Validate ritual ID (basic validation - positive number only)
+   * @param {DomainName} domain - Domain name (unused but kept for API compatibility)
+   * @param {number} ritualId - Ritual ID to validate
+   * @returns {boolean} True if valid (positive number)
+   */
+  static isValidRitualId(domain: DomainName, ritualId: number): boolean {
+    return typeof ritualId === 'number' && ritualId > 0;
+  }
+
+  /**
+   * Validate provider compatibility with domain
+   * @param {DomainName} domain - Domain name
+   * @param {PublicClient | ethers.providers.Provider} provider - Provider to validate (viem PublicClient or ethers Provider)
+   * @returns {Promise<boolean>} True if provider is valid for domain
+   */
+  static async isValidProvider(
+    domain: DomainName,
+    provider: PublicClient | ethers.providers.Provider,
+  ): Promise<boolean> {
+    let chainId: number;
+
+    if (!provider || typeof provider !== 'object') {
+      // Invalid provider
+      return false;
+    }
+
+    // Try to detect provider type and get chain ID safely
+    try {
+      // Check if it's a viem PublicClient (has getChainId method)
+      if (
+        'getChainId' in provider &&
+        typeof provider.getChainId === 'function'
```

Comment:
I thought there were helper methods to check whether viem client or ethers provider?